generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationOutputValidator.java

```diff
@@ -1,116 +1,58 @@
 package com.scottlogic.deg.generator.validators;
 
 import com.google.inject.Inject;
+import com.google.inject.name.Named;
 import com.scottlogic.deg.generator.Profile;
-import com.scottlogic.deg.generator.generation.GenerationConfigSource;
+import com.scottlogic.deg.generator.commandline.OutputTargetSpecification;
 import com.scottlogic.deg.generator.inputs.validation.Criticality;
 import com.scottlogic.deg.generator.inputs.validation.ProfileValidator;
 import com.scottlogic.deg.generator.inputs.validation.ValidationAlert;
 import com.scottlogic.deg.generator.inputs.validation.ValidationType;
 import com.scottlogic.deg.generator.inputs.validation.messages.OutputValidationMessage;
-import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
-import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
-import com.scottlogic.deg.generator.utils.FileUtils;
+import com.scottlogic.deg.generator.outputs.targets.OutputTargetValidationException;
+import com.scottlogic.deg.generator.outputs.targets.ValidatableOutput;
 
 import java.io.IOException;
-import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 public class GenerationOutputValidator implements ProfileValidator {
-    private final FileUtils fileUtils;
-    private final GenerationConfigSource configSource;
-    private final OutputTarget outputTarget;
+    private final OutputTargetSpecification outputTarget;
+    private final boolean inViolationMode;
 
     @Inject
-    public GenerationOutputValidator(FileUtils fileUtils, GenerationConfigSource configSource, OutputTarget outputTarget) {
-        this.fileUtils = fileUtils;
-        this.configSource = configSource;
+    GenerationOutputValidator(
+        OutputTargetSpecification outputTarget,
+        @Named("config:inViolationMode") boolean inViolationMode) {
+
+        // TODO: Change this! The validatee shouldn't be a dependency of the validator
```

Comment:
Is this something that should be changed before merging or is the intention to do this work as part of stdout feature

Reply:
That issue existed before my PR - I just added the comment to call it out. I might address it as part of the stdout work, but they're not directly related and changing it might involve some rippling effects elsewhere. (plus might require some discussion of good DI philosophy)

Reply:
i'd prefer we raise a github issue rather than a todo

Reply:
There's a school of thought that says issue backlogs are where good intentions go to die :sweat_smile: I don't mind that approach, but I'd want something in the code in the meantime, to break people's pattern-copying instincts and nudge them away from committing the same sin.

Reply:
both?